tasks/installation.yml

```diff
@@ -176,6 +181,20 @@
     - magento_release_updated.changed
     - magento_skip_release == False
 
+- name: "Check if Magento database exists"
+  shell:
+    cmd: |
+      mysql --user="{{ magento_config_db.username }}" --password="{{ magento_config_db.password }}" --database="{{ magento_config_db.name }}" --host="{{ magento_config_db.host }}" <<EOF
+        SHOW TABLES LIKE 'core_config_data';
+      EOF
+  register: magento_db_status
+
+- name: "Create the Magento database if missing"
+  become: "yes"
+  become_user: "{{ magento_user }}"
+  command: "/usr/bin/php {{ magento_app_root }}/bin/magento setup:db-schema:upgrade"
+  when: magento_db_status.stdout_lines|length == 0
+
```

Comment:
These two steps look a bit fishy. I know chances are rather slim but what if for any reason database is corrupted? I don't even mean scenario when just `core_config_data` exists because of someone deleted rest... What I am actually worried about is incomplete import. Core_config_data will be in first half of the dump usually - if import fails variable "magento_db_status" will actually hold false positive and thus block re-import...

Reply:
that is valid observation, at the same time we are not re-importing the database, we are creating it from scratch and only if it is not there in any form.

If database is there in any way - we don't attempt to create it, cause otherwise there will be unexpected side-effects.

If database is there but corrupted, deploy will fail on running database migration steps, which is fine and the best that ansible can do in that situation (if it ever happens).